environment/src/main/java/com/sequenceiq/environment/network/service/SubnetIdProvider.java

```diff
@@ -24,12 +29,41 @@
     private CloudPlatformConnectors cloudPlatformConnectors;
 
     public String provide(NetworkDto network, Tunnel tunnel, CloudPlatform cloudPlatform) {
-        LOGGER.debug("Choosing subnet, network: {},  platform: {}, tunnel: {}", network, cloudPlatform, tunnel);
         if (network == null || network.getSubnetIds() == null || network.getSubnetIds().isEmpty() || network.getCbSubnets() == null
                 || network.getCbSubnets().isEmpty()) {
             LOGGER.debug("Check failed, returning null");
             return null;
         }
+        return selectSubnet(network, tunnel, cloudPlatform, network.getCbSubnetValues(), null);
+    }
+
+    public String provideEndpointGateway(NetworkDto network, CloudPlatform cloudPlatform, String baseSubnetId) {
+        if (baseSubnetId == null || network == null ||
+                ((network.getEndpointGatewaySubnetMetas() == null || network.getEndpointGatewaySubnetMetas().isEmpty())
+                && (network.getCbSubnets() == null || network.getCbSubnets().isEmpty()))) {
+            LOGGER.debug("Check failed, returning null");
+            return null;
+        }
+        // Use tunnel type DIRECT so that any logic that checks if CCM/private subnets are required evaluates to false
```

Comment:
CCM is a parameter on environment request, is there any validation that this feature and CCM could not be enabled at the same time??
If not and a customer has enabled load balancer feature then the CCM option doesn't make sense on the wizards.

Reply:
@biharitomi the comments on this class belong in my PR. Enis just got them from rebasing on an old version of that PR.

@enis there have been significant changes since you last rebased. You should rebase again.

Reply:
Per the comment, I don't think there's any reason we can't enable CCM and this feature at the same time. That's really the whole point of the feature; the customer can have a fully private environment that's only accessible inside their VPC, but they want public UI and API endpoints so that the UI and API can be called by jobs outside of their VPC. In that case we still need a way to tell CB that the environment itself is private and should use CCM.

Reply:
Indeed these are due to the fact that the branch that this PR was based on got changed. Please take another look since it is a much smaller patch.